cnudie/retrieve.py

```diff
@@ -37,6 +41,82 @@
     typing.Sequence[str]
 ]
 
+OcmRepositoryCfg = \
+    str | typing.Iterable[str] | \
+    cnudie.util.OcmLookupMappingConfig | \
+    cnudie.util.OcmLookupMapping | \
+    typing.Iterable[cnudie.util.OcmLookupMapping]
+
+
+OcmRepositoryLookup = typing.Callable[
+    [ComponentName],
+    typing.Generator[cm.OciRepositoryContext | str, None, None],
+]
+
+
+def _iter_ocm_repositories(
+    component: str | cm.ComponentIdentity | cm.Component,
+    repository_cfg: OcmRepositoryCfg,
+    /,
+):
+    if isinstance(component, cm.ComponentIdentity):
+        component = component.name
+    elif isinstance(component, cm.Component):
+        component = component.name
+
+    if isinstance(repository_cfg, cm.OciRepositoryContext):
+        yield repository_cfg
+        return
+
+    if isinstance(repository_cfg, str):
+        yield repository_cfg
+        return
+
+    if isinstance(repository_cfg, cnudie.util.OcmLookupMapping):
+        repository_cfg: cnudie.util.OcmLookupMapping
+        if repository_cfg.matches(component):
+            yield repository_cfg.ocm_repo_url
+        return
+
+    if isinstance(repository_cfg, cnudie.util.OcmLookupMappingConfig):
+        yield repository_cfg.iter_ocm_repositories(component)
```

Comment:
We should rather `yield from` as `iter_ocm_repositories` returns a Generator.

Reply:
this will not work, as `_iter_ocm_repositories` is expected to return an iterable in `iter_ocm_reposositories`